File: web/src/common/type/article.ts
import {TComment} from './comment';

export type TArticle = {
  articleId: string,
  title: string,
  content: string,
  comments: TComment[],
  createTime: string,
  // eslint-disable-next-line camelcase
  modify_time: string,
  // eslint-disable-next-line camelcase
  is_private: number,
  type: string,
  username: string,
  avatar: string,
}

File: server/src/main/java/com/example/server/entity/Article.java
package com.example.server.entity;

import lombok.Data;

import java.util.Date;
import java.util.List;

@Data
public class Article {
    private int id;
    private String title;
    private String content;
    private String author_id;
    // 是否匿名  0 -- 公开  1 -- 匿名
    private int is_private;
    private Date create_time;
    private Date modify_time;
    private List<Comment> comments;
    private String username;
    private String type;
    private String avatar;
}
